### PR TITLE
Add DependOn, DeletionPolicy and others to CustomResource

### PR DESCRIPTION
**Pull request by xrn**
Solving #270

### CODE REVIEW

**Comment by xrn** [comment]
@PaulMaddox will be awesome if you can take a look

**Comment by PaulMaddox** [comment]
Thanks @xrn - awesome as always. 

Could you add a test case that covers this?

File: goformation_test.go [hunk 1]
```diff
@@ -249,13 +249,6 @@ var _ = Describe("Goformation", func() {
 			Expect(template).ShouldNot(BeNil())
 		})
 
-		resources := template.GetAllCustomResources()
-
-		It("should have exactly one resource", func() {
-			Expect(resources).To(HaveLen(1))
-			Expect(resources).To(HaveKey("MyCustomResource"))
-		})
-
```
Comment (PaulMaddox) [inline goformation_test.go:258]: What's the reason for removing this test?

**Review by PaulMaddox — CHANGES_REQUESTED** [review]

File: goformation_test.go (hunk 1 above)
Comment (xrn) [reply goformation_test.go:258]: 2 reasons 

1. I removed `GetAllCustomResources()` function seems that it do not make bigger sense to have it - then implementation of a resource is not necessarily different that others 
2. In the next lines (269-271) we can find 

resources, ok := result["Resources"].(map[string]interface{})
Expect(ok).To(BeTrue())
Expect(resources).To(HaveLen(1))
Expect(resources).To(HaveKey("MyCustomResource"))
			
Which is in my opinion almost the same test but without - template.GetAllCustomResources() - this is why I think we can remove that part of the test

**Comment by xrn** [comment]
Maybe a bit explanation what I did here 

1. I copied 100% of any other resource like for example - https://github.com/awslabs/goformation/blob/master/cloudformation/cognito/aws-cognito-identitypool.go
2. I change it to 

```
type CustomResource struct {
	Type       string                 `json:"Type,omitempty"`
	Properties map[string]interface{} `json:"Properties,omitempty"`
```

3. I moved from `Properties:          (Properties)(r),` to `Properties:          (map[string]interface{})(r.Properties),` that to what CustomResource is almost identical as other resources

**Pull request merged**